File: daily_python/3741_Minimum_Distance_Between_Three_Equal_Elements_II.py
import os
import sys
from typing import List
from collections import defaultdict

class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        m = defaultdict(list)
        ans = float('inf')
        for i in range(n):
            m[nums[i]].append(i)
        for v in m.values():
            if len(v) >= 3:
                for i in range(2, len(v)):
                    ans = min(ans, 2 * (v[i] - v[i-2]))
        return ans if ans != float('inf') else -1

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.minimumDistance(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/3741_Minimum_Distance_Between_Three_Equal_Elements_II.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minimumDistance(vector<int>& nums) {
        int n = nums.size();
        int ans = INT_MAX;
        vector<vector<int>> table(n + 1);
        for (int i = 0; i < n; i++){
            table[nums[i]].push_back(i);
        }
        for (auto& v : table){
            if (v.size() >= 3){
                for (int i = 2; i < v.size(); i++){
                    ans = min(ans, (v[i] - v[i-2]) * 2);
                }
            }
        }
        return (ans == INT_MAX)? -1 : ans;
    }
};

int main(){
    vector<int> nums = {1,2,1,1,3};
    // vector<int> nums = {1,1,2,3,2,1,2};
    // vector<int> nums = {1};
    Solution S;
    cout << S.minimumDistance(nums) << endl;
    return 0;
}